TypeSystem/flow-analysis/reachability_A22_t01.dart

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion - Method invocation: If `N` is an expression of the form
+/// `E1.m1(E2)`, then:
+/// - Let `before(E1) = before(N)`
+/// - Let `before(E2) = after(E1)`
+/// - Let `T` be the static return type of the invocation
+/// - If `T <: Never` then:
+///   - Let `after(N) = unreachable(after(E2))`.
+/// - Otherwise:
+///   - Let `after(N) = after(E2)`.
+///
+/// @description Checks that for an expression of the form `E1.m1(E2)`
+/// `before(E2) = after(E1)`.
```

Comment:
At this location in the feature spec we have a couple of 'TODO' comments. For example, there's no mention of the case where `m1` is a getter whose return type is a function type or a interface type that includes a method named `call` ... or `Never`, which would allow the invocation `E1.m1(E2)` as well.

I think it would make sense to have those cases as well. When the feature spec gets expanded to cover those cases we'll just need to revisit these tests and check that they aren't expecting anything that contradicts the newly added spec cases.

For this PR, I'll just assume that we're calling a method.

Reply:
Regarding "checks that .. `before(E2) = after(E1)`", that's a big ask. Perhaps just say that we're checking one consequence of that property (namely that if `after(E1)` is unreachable then `before(E2)` is also unreachable)?

Reply:
I had a couple of other comments, so it might make sense to add some extra tests to this PR&mdash;whatever is more convenient.